rts/lra/coordinator/src/main/java/io/narayana/lra/coordinator/internal/LRARecoveryModule.java

```diff
@@ -75,7 +76,7 @@ public void periodicWorkFirstPass() {
         // uids per transaction type
         InputObjectState aa_uids = new InputObjectState();
 
-        if (getUids(aa_uids)) {
+        if (getUids(aa_uids, _transactionType)) {
```

Comment:
Actually, here the problem was, if there is a failed record present then, it would not be moved to failed location until recoveryManager executes its next run, so before that we need to check both the locations.

Reply:
My point was that the requirement for this new functionality is twofold: to move the failed record and to query for failed transactions by looking in a specific part of the store.

Reply:
okay